### PR TITLE
Handle missing name parts from ORCID.

### PR DESCRIPTION
**Pull request by thewilkybarkid**
@giorgiosironi found this last week.

### CODE REVIEW

File: test/commands/test_update_profile_from_orcid_record.py [hunk 1]
```diff
@@ -52,6 +52,39 @@ def test_it_updates_the_name():
...
+def test_it_does_not_updates_the_name_if_its_missing():
+    profile = Profile('12345678', Name('Old Name'))
+    orcid_record = {'person': {'name': {}}}
+
+    update_profile_from_orcid_record(profile, orcid_record)
+
+    assert profile.name == Name('Old Name')
+
+
+def test_it_updates_the_name_if_there_is_no_given_name():
```
Comment (thewilkybarkid) [inline test/commands/test_update_profile_from_orcid_record.py]: Through the ORCID UI it appears to be impossible to trigger this (a private name seems not to have the `name` section at all). _But_, the XSD seems to say that it is possible. 🤷‍♂️

**Comment by giorgiosironi** [comment]
Some samples from the public dataset, to see if any corner case comes to mind: [orcid_sample_1000_names.txt](https://github.com/elifesciences/profiles/files/1505373/orcid_sample_1000_names.txt)

**Comment by giorgiosironi** [comment]
From the samples:

- UTF-8 handling in increasing level of difficult `Francisco / Baños`, `Verónica / López López`, `Татьяна / Яковлева`, `璞玉 / 田` 
- possibly unexpected characters `Francisco Javier / Cuevas-de-la-Rosa`
- people writing without capitals `jian / yilang`
- PEOPLE WRITING IN UPPERCASE `MARIA ISABEL  / GONZALEZ SANCHEZ`

**Comment by thewilkybarkid** [comment]
Time for some property tests...

**Comment by giorgiosironi** [comment]
Yes, but actual data from the service is higher quality than generating random strings

**Pull request merged**